Library/Homebrew/cmd/uninstall.rb

```diff
@@ -26,6 +53,36 @@ def uninstall
         name = Formulary.canonical_name(name)
         rack = HOMEBREW_CELLAR/name
 
+        # Cellar/oldname can be a symlink to Cellar/newname so in order not to
+        # get newname's keg as oldname's keg we check if it's a symlink.
+        if rack.symlink?
+          raise "#{rack.to_s} is a symlink"
+        end
```

Comment:
Why half of files? It doesn't uninstall anything if rack is a symlink. The same functionality was added to ARGV, so uninstall doesn't uninstall half of files.

But yes, maybe it worth raising at the very beginning:

```ruby
ARGV.named.each do |name|
  if HOMEBREW_CELLAR.join(name).symlink?
    raise "#{HOMEBREW_CELLAR/name} is a symlink"
  end
end
```